### PR TITLE
[FIX authentication] restore auth signup modal on articles

### PR DESCRIPTION
**Pull request by xtina-starr**
Addresses https://artsyproduct.atlassian.net/browse/GROW-887. We lost the auth modal trigger when we removed the old [editorial sign up view](https://github.com/artsy/force/commit/31130073685a018bce2b5e90fce8c40f865284b7). This restores the auth modal by moving the modal trigger to the `componentDidMount` function of the `<Article>` component.

### CODE REVIEW

**Comment by artsyit** [comment]
Deploy preview for *artsy-reaction* ready!

Built with commit 9d12e00e0c278b2358bce705a54e3a84655e86b6

https://deploy-preview-1230--artsy-reaction.netlify.com

File: src/Components/Publishing/Article.tsx [hunk 1]
```diff
@@ -50,6 +51,28 @@ export interface ArticleProps {
   }
 )
 export class Article extends React.Component<ArticleProps> {
+  componentDidMount() {
+    if (!this.props.isLoggedIn && window) {
```
[the remark this answers is not in the document]
Comment (xtina-starr) [reply src/Components/Publishing/Article.tsx]: @damassi to make sure I understand, you're saying that the window check isn't needed in this case but in the future when there is a legit need to check for the present of the window, I should do so via `typeof window !== 'undefined'`?

[the remark this answers is not in the document]
Comment (damassi) [reply src/Components/Publishing/Article.tsx]: Yup yup :) `componentDidMount` will only ever execute on the client, which presumes a `window`. But in a different case where you're not sure of the context and want to be safe can check like...

```jsx
const Article = props => {
  const isSever = typeof window === 'undefined'

  if (isServer) {
    return <PlaceholderBlockText /> 
  } else {
   return <ArticleLoader /> 
  }
}
```  

If `window` isn't guarded behind a `typeof` check the Node env would encounter it, think it was a common variable, and then error out as undefined.

[the remark this answers is not in the document]
Comment (xtina-starr) [reply src/Components/Publishing/Article.tsx]: Yeah that makes sense. Thanks for clarifying 😄

**Pull request merged**